examples/maxq10xx-wolfssl-client.c

```diff
@@ -0,0 +1,335 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* This example shows how to write a simple TLS client that uses the features
+ * features of the Analog Devices MAXQ 1065 and 1080. */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+/* wolfSSL */
+#ifndef WOLFSSL_USER_SETTINGS
+    #include <wolfssl/options.h>
+#endif
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/sha256.h>
+#include <wolfssl/wolfcrypt/cryptocb.h>
+#include <wolfssl/wolfcrypt/ecc.h>
+#include <wolfssl/wolfcrypt/rsa.h>
+#include <wolfssl/wolfcrypt/asn.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+
+#if (defined(WOLFSSL_MAXQ1065) || defined(WOLFSSL_MAXQ108x)) && \
+    defined(HAVE_PK_CALLBACKS) && defined(WOLF_CRYPTO_CB) && \
+    !defined(NO_PSK) && !defined(HAVE_EXTENDED_MASTER) && \
+    defined(NO_WOLFSSL_SERVER)
+
+/* -------------- */
+/* Configurations */
+/* -------------- */
+
+/* Please define this if you want wolfSSL's debug output */
+#define WANT_DEBUG
+
+/* Please set the server's address and the port it listens on */
+#define DEFAULT_SERVER "127.0.0.1"
+#define DEFAULT_PORT   11111
+
+/* Please define one of USE_ECDHE_ECDSA, USE_FFDHE_RSA, USE_PSK */
+//#define USE_ECDHE_ECDSA
+//#define USE_FFDHE_RSA
+#define USE_PSK
+
+/* Please define one of USE_TLSV13, USE_TLSV12 */
+//#define USE_TLSV13
+#define USE_TLSV12
+
+/* Please set the location of the dummy private keys */
+#if defined(USE_ECDHE_ECDSA) || defined(USE_PSK)
+#define KEYPUB_FILE "/home/anthony/ecc-p256-pub.pem"
+#else
+#define KEYPUB_FILE "/home/anthony/rsa-2048-pub.pem"
+#endif
+
+/* ------------------------------------ */
+/* No modifications required below here */
+/* ------------------------------------ */
+
+#ifdef USE_PSK
+
+/* identity is OpenSSL testing default for openssl s_client, keep same */
+static const char* kIdentityStr = "Client_identity";
+
+static WC_INLINE unsigned int my_psk_client_cb(WOLFSSL* ssl, const char* hint,
+        char* identity, unsigned int id_max_len, unsigned char* key,
+        unsigned int key_max_len)
+{
+    (void)ssl;
+    (void)hint;
+    (void)key_max_len;
+
+    /* see internal.h MAX_PSK_ID_LEN for PSK identity limit */
+    XSTRNCPY(identity, kIdentityStr, id_max_len);
+
+    if (wolfSSL_GetVersion(ssl) < WOLFSSL_TLSV1_3) {
+        /* test key in hex is 0x1a2b3c4d , in decimal 439,041,101 , we're using
+           unsigned binary */
+        key[0] = 0x1a;
+        key[1] = 0x2b;
+        key[2] = 0x3c;
+        key[3] = 0x4d;
+
+        return 4;   /* length of key in octets or 0 for error */
+    }
+    else {
+        int i;
+        int b = 0x01;
+
+        for (i = 0; i < 32; i++, b += 0x22) {
+            if (b >= 0x100)
```

Comment:
I don't really understand the logic of this loop. Explanatory comment would help.

Reply:
This is taken from test.h. Would prefer to keep it the same.